src/mobx.ts

```diff
@@ -35,17 +36,25 @@ export let events = <T extends { new(...args: any[]): any }>(target: T): T => {
     }[className];
 }
 
-function isObservableEvent(o: any): o is IObservableEvent<any, any> {
-    return typeof o == 'function' && !!o.subscribe;
-}
+function wrapAsync(name: string, async: any): any {    
+    if(isObservableEvent(async)) {
+        return Object.assign(Object.create(async), {
+            subscribe(next: (value: any) => void, error?: (err: any) => void, complete?: () => void): ISubscription {
+                return wrapAsync(name, async.subscribe(

```

Comment:
I don't think this wrapAsync is necessary, since async.subscribe returns a subscription, not another async type